django/core/management/commands/inspectdb.py

```diff
@@ -69,7 +76,15 @@ def strip_prefix(s):
                         constraints = connection.introspection.get_constraints(cursor, table_name)
                     except NotImplementedError:
                         constraints = {}
-                    primary_key_column = connection.introspection.get_primary_key_column(cursor, table_name)
+                    except ValueError:
+                        # this might have failed because of sqlite3 handling of views
```

Comment:
So this is a branch that only SQLite hits? That doesn't seem ideal -- inspectdb should be mostly database agnostic. Can that logic go in SQLite's `get_constraints()`?

Reply:
I've now moved the SQLite-specific code into the SQLite backend as you suggest. It does make the main inspectdb code much neater. I've left it as a separate commit so you can check the changes, hope it helps.